packages/flutter/lib/src/cupertino/search_field.dart

```diff
@@ -0,0 +1,412 @@
+  ///
+  /// Defaults to clearing the text. The suffix action is customizable
+  /// so that users can override it with other functionality, that isn't
+  /// necessarily clearing text.
+  final VoidCallback? onSuffixTap;
+
+  /// {@macro flutter.material.textfield.restorationId}
+  final String? restorationId;
+
+  /// {@macro flutter.widgets.Focus.focusNode}
+  final FocusNode? focusNode;
+
+  @override
+  State<StatefulWidget> createState() => _CupertinoSearchTextFieldState();
+}
+
+class _CupertinoSearchTextFieldState extends State<CupertinoSearchTextField>
+    with
+        RestorationMixin,
+        AutomaticKeepAliveClientMixin<CupertinoSearchTextField> {
+  /// Default value for the border radius. Radius value was determined using the
+  /// comparison tool in https://github.com/flutter/platform_tests/.
+  final BorderRadius _kDefaultBorderRadius =
+      const BorderRadius.all(Radius.circular(9.0));
+
+  RestorableTextEditingController? _controller;
+
+  TextEditingController get _effectiveController =>
+      widget.controller ?? _controller!.value;
+
+  FocusNode? _focusNode;
+
+  FocusNode get _effectiveFocusNode =>
+      widget.focusNode ?? (_focusNode ??= FocusNode());
+
+  @override
+  void initState() {
+    super.initState();
+    if (widget.controller == null) {
+      _createLocalController();
+    }
+  }
+
+  @override
+  void didUpdateWidget(CupertinoSearchTextField oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.controller == null && oldWidget.controller != null) {
+      _createLocalController(oldWidget.controller!.value);
+    } else if (widget.controller != null && oldWidget.controller == null) {
+      unregisterFromRestoration(_controller!);
+      _controller!.dispose();
+      _controller = null;
+    }
+  }
+
+  @override
+  void restoreState(RestorationBucket? oldBucket, bool initialRestore) {
+    if (_controller != null) {
+      _registerController();
+    }
+  }
+
+  void _registerController() {
+    assert(_controller != null);
+    registerForRestoration(_controller!, 'controller');
+    _controller!.value.addListener(updateKeepAlive);
+  }
+
+  void _createLocalController([TextEditingValue? value]) {
+    assert(_controller == null);
+    _controller = value == null
+        ? RestorableTextEditingController()
+        : RestorableTextEditingController.fromValue(value);
+    if (!restorePending) {
+      _registerController();
+    }
+  }
+
+  @override
+  String? get restorationId => widget.restorationId;
+
+  @override
+  bool get wantKeepAlive => _controller?.value.text.isNotEmpty == true;
+
+  @override
+  Widget build(BuildContext context) {
+    super.build(context); // See AutomaticKeepAliveClientMixin.
+
+    final String placeholder = widget.placeholder ??
+        CupertinoLocalizations.of(context)?.searchTextFieldPlaceholerLabel ??
+        'Search';
+
+    final TextStyle placeholderStyle = widget.placeholderStyle ??
+        TextStyle(
```

Comment:
nit: I think `CupertinoTextField` should be able to resolve the color correctly even if you just say 
```dart
    final TextStyle placeholderStyle = widget.placeholderStyle ?? TextStyle(color: CupertinoColors.secondaryLabel);
```